core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -209,6 +210,42 @@ class ZkUtils(val zkClient: ZkClient,
     }
   }
 
+
+  def getClusterId(): Option[String] = {
+    def parseClusterId(clusterIdString: String): String = {
+
+      Json.parseFull(clusterIdString) match {
+        case Some(m) =>
+          val clusterIdMap = m.asInstanceOf[Map[String, Any]]
+          clusterIdMap.get("id").get.asInstanceOf[String]
+        case None => throw new KafkaException("Failed to parse the cluster id json [%s].".format(clusterIdString))
+      }
```

Comment:
You can use `map/getOrElse` instead of this. Something like:

```scala
Json.parseFull(clusterIdString).map { m =>
  ...
}.getOrElse(throw new KafkaException(...))
```

Also, use string interpolation (`s"$foo"`) instead of `format`.